public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx

```diff
@@ -154,7 +160,7 @@ export class ConfigEditor extends PureComponent<Props, State> {
                   <Input
                     className="width-30"
                     placeholder="default"
-                    value={profile}
+                    value={options.jsonData.profile}
```

Comment:
We are actually not using the profile from state that we've set in `componentDidMount` here. So to use the profile from state, it should be `value={profile}` and  on line 124, we can do `const { regions, profile } = this.state;`.

Reply:
thank you for your feedback!

I realised that the nature of the problem was just the condition `if (!profile)`
which would become true even if the user deletes completely the input.

So I reverted the changes with the state and I just changed the condition
to fallback to the old field (`database`) only if the new field (`profile`) is undefined.

Would you like to check it once again?